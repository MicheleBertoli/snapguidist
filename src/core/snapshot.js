const { SnapshotState } = require('jest-snapshot')
const path = require('path')

const base = './.snapguidist/__snapshots__/'
const typeOf = { value: Symbol.for('react.test.json') }

const snapshot = (name, tree, update) => {
  const destination = path.resolve(base, `${name}.snap`)
  const state = new SnapshotState(null, update, destination)

  Object.defineProperty(tree, '$$typeof', typeOf)

  const result = state.match(name, tree)
  state.save(update)

  return result
}

module.exports = snapshot
