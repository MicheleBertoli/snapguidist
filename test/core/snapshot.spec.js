import fs from 'fs-extra'
import snapshot from '../../src/core/snapshot'

beforeEach(() => {
  fs.emptyDirSync('./.snapguidist/__snapshots__')
})

afterAll(() => {
  fs.removeSync('./.snapguidist')
})

test('passes if new', () => {
  const tree = { type: 'div' }
  const result = snapshot('name', tree)

  expect(result).toEqual({ pass: true })
})

test('fails if the type changes', () => {
  const tree = { type: 'div', children: [{ type: 'span' }] }
  snapshot('name', tree)

  tree.type = 'span'
  const result = snapshot('name', tree)

  expect(result).toMatchSnapshot()
})

test('does not fail if update is true', () => {
  const tree = { type: 'div' }
  snapshot('name', tree)

  tree.type = 'span'
  const update = true
  const result = snapshot('name', tree, update)

  expect(result).toEqual({ pass: true })
})
