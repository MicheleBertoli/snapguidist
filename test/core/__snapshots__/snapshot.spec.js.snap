// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fails if the type changes 1`] = `
Object {
  "actual": "<span>
  <span />
</span>",
  "count": 1,
  "diff": "- Expected
+ Received


- <div>
+ <span>
    <span />
- </div>
+ </span>",
  "expected": "<div>
  <span />
</div>",
  "pass": false,
}
`;
